node/resolvers/Mutations/Users.ts

```diff
@@ -249,11 +253,20 @@ const Users = {
 
     sendImpersonateB2BUserMetric(ctx, metricParams)
 
-    return (
-      impersonation?.data?.impersonateUser ?? {
-        status: 'error',
-      }
-    )
+    const result = impersonation?.data?.impersonateUser ?? { status: 'error' }
+
+    await audit.sendEvent({
+      subjectId: 'impersonate-b2b-user-event',
+      operation: 'IMPERSONATE_B2B_USER',
+      meta: {
+        entityName: 'B2BUser',
```

Comment:
I think we are always talking about an "User" from an organization here, so maybe lets keep the entity just as "User" across the app in cases like this.

```suggestion
        entityName: 'User',
```